packages/wdio-browserstack-service/src/launcher.ts

```diff
@@ -45,11 +47,14 @@ export default class BrowserstackLauncherService implements Services.ServiceInst
                     } else if (shouldAddServiceVersion(this._config, this._options.testObservability)) {
                         capability['browserstack.wdioService'] = bstackServiceVersion
                     }
+                    this._buildIdentifier = capability['browserstack.buildIdentifier']?.toString()
+                    this._buildName = capability['build']?.toString()
```

Comment:
not sure why we need `toString()` here

```suggestion
                    this._buildIdentifier = capability['browserstack.buildIdentifier']
                    this._buildName = capability.build
```

Reply:
We were getting TypeError, `Type 'number' is not assignable to type 'string'`. That's why, had to cast it. 

We have added tests regarding the functionality, which part we want more tests to be added more specifically ?

Reply:
> We were getting TypeError, `Type 'number' is not assignable to type 'string'`. That's why, had to cast it.

Ah ok, understand